akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala

```diff
@@ -377,16 +377,8 @@ object ForkJoinExecutorConfigurator {
                                threadFactory: ForkJoinPool.ForkJoinWorkerThreadFactory,
                                unhandledExceptionHandler: Thread.UncaughtExceptionHandler)
     extends ForkJoinPool(parallelism, threadFactory, unhandledExceptionHandler, true) with LoadMetrics {
-    override def execute(r: Runnable): Unit = {
-      if (r eq null) throw new NullPointerException("The Runnable must not be null")
-      val task =
-        if (r.isInstanceOf[ForkJoinTask[_]]) r.asInstanceOf[ForkJoinTask[Any]]
```

Comment:
Because your whole commit has been reverted. We don't want to take risks on 2.3.x anymore, this bug is critical already. The original code worked for many-many versions in a row. If we want to experiment further, it will go into 2.4, but 2.3 remains conservative in this regard.